src/components/ui/TabButtonGroup/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { View } from "react-native";
+import { Button } from "react-native-elements";
+import { RFValue } from "react-native-responsive-fontsize";
+import styles from "./styles";
+import { colors } from "@app/utils";
+
+interface TabButtonGroupProps {
+  id?: string;
+  onPress: (index: number) => void;
+  selectedIndex: number;
+  buttons: string[];
+}
+
+const TabButtonGroup: React.FC<TabButtonGroupProps> = ({
+  id,
+  onPress,
+  selectedIndex,
+  buttons,
+}) => {
+  return (
+    <View data-testid={id} style={styles.container}>
+      {buttons.map((button, index) => {
+        const selected = selectedIndex === index;
+        return (
+          <Button
+            key={`${index}-${button}`}
+            data-testid={`${index}-${button}-btn`}
+            title={button}
+            onPress={() => onPress(index)}
+            type={selected ? "solid" : "outline"}
+            containerStyle={{
+              marginRight: RFValue(10),
+              
+            }}
+            buttonStyle={{ backgroundColor: selected ? colors.darkBrown : colors.white, borderColor: colors.darkBrown }}
```

Comment:
how come you aren't defining styles in styles.ts

Reply:
According to these sources, you can pass props into the styles variable by making it into a callback function

https://stackoverflow.com/questions/42707327/passing-props-into-external-stylesheet-in-react-native
https://cloudstack.ninja/arahpanah/how-to-pass-props-to-stylesheet-in-react-native-using-typescript/

Reply:
buttonStyle is a prop from Button. I can try just passing style, but not sure if it'll work